compiler/quilt/nodes.py

```diff
@@ -139,6 +140,38 @@ def _data(self):
 
         return store.load_dataframe(hash_list)
 
+def _create_filter_func(filter_dict):
+    filter_name = filter_dict.pop('name', None)
+    if filter_name is not None and not isinstance(filter_name, string_types):
+        raise ValueError("Invalid 'name': %r" % filter_name)
+
+    filter_meta = filter_dict.pop('meta', None)
+    if filter_meta is not None and not isinstance(filter_meta, dict):
+        raise ValueError("Invalid 'meta': %r" % filter_meta)
+
+    if filter_dict:
+        raise ValueError("Unexpected data in the filter: %r" % filter_dict)
```

Comment:
I saw this error and didn't know what to do. Maybe include the top-level keys in the error message?